src/workspace-data/data-table/uri-viewer/UriPreview.js

```diff
@@ -22,23 +22,24 @@ const styles = {
   },
 };
 
-const isImage = ({ contentType, name }) => {
-  return /^image/.test(contentType) || /\.(?:(jpe?g|png|svg|bmp))$/.test(name);
+export const isImage = ({ contentType, name }) => {
+  return /^image/.test(contentType) || /\.(jpe?g|png|svg|bmp)$/.test(name);
 };
 
-const isText = ({ contentType, name }) => {
-  return /(?:(^text|application\/json))/.test(contentType) || /\.(?:(txt|[ct]sv|log|json))$/.test(name);
+export const isText = ({ contentType, name }) => {
+  return /(^text|application\/json)/.test(contentType) || /\.(txt|[ct]sv|log|json|fastq|fasta|fa|vcf|sam|bed|interval_list|gtf|md|html)$/.test(name);
 };
 
-const isBinary = ({ contentType, name }) => {
-  // excluding json and google's default types (i.e. wasn't set to anything else)
-  return (
-    /application(?!\/(json|octet-stream|x-www-form-urlencoded)$)/.test(contentType) || /(?:(\.(?:(ba[mi]|cra[mi]|pac|sa|bwt|gz))$|\.gz\.))/.test(name)
-  );
+export const isBinary = ({ contentType, name }) => {
+  return /application(?!\/(json|octet-stream|x-www-form-urlencoded)$)/.test(contentType) || /(\.(ba[mi]|cra[mi]|bcf|h5ad|pdf|gz)$|\.gz\.)/.test(name);
 };
 
-const isFilePreviewable = ({ size, ...metadata }) => {
-  return !isBinary(metadata) && (isText(metadata) || (isImage(metadata) && size <= 1e9));
+export const canRender = ({ contentType, name }) => {
+  return /(^text\/html|application\/pdf)/.test(contentType) || /\.(html|pdf)$/.test(name);
+};
+
+export const isFilePreviewable = ({ size, ...metadata }) => {
+  return isBinary(metadata) || isText(metadata) || (isImage(metadata) && size <= 1e9);
```

Comment:
Enabling FilePreview for binary, text, and image files that do not exceed 1GB.

Reply:
what happens when previewing binary files? Is the preview output usable? I would expect it to be unintelligible. Like this preview of a `.gz` file:
![Screenshot 21-08-2024 at 09 13](https://github.com/user-attachments/assets/79f0c962-8306-4c4f-94b2-8ff3508d45a8)